lib/AST/ASTGates.cpp

```diff
@@ -1131,7 +1131,13 @@ ASTGateNode::ASTGateNode(const ASTIdentifierNode *Id,
     ++C;
   }
 
-  assert(C == Params.size() && "Inconsistent number of Params!");
+  if (C != Params.size()) {
+    std::stringstream M;
+    M << "Inconsistent number of Params." << C;
+    QasmDiagnosticEmitter::Instance().EmitDiagnostic(
+        DIAGLineCounter::Instance().GetLocation(), M.str(), DiagLevel::Error);
```

Comment:
Could we maybe update the message to something like `Inconsistent number of parameters in the gate call for the corresponding gate definition`? 

There is also a similar diagnostic further up that should be edited.

Reply:
Updated in https://github.com/openqasm/qe-qasm/pull/34/commits/3a1c35975ab7e3293a452dea5a3c8223688666ac